### PR TITLE
Fix/178 rating

### PR DESCRIPTION
**Pull request by tetogomez**
### Rate BP page #178 #179 

- get data from the blockchain (EOS JS API)
- add eosjs package
- create a function to get color using account Name (this way we avoid repeat colors)
- validate user rating data
- handle empty bp objects
- populate rating slider data
- add a message if user rating a bp without login
- add a mutation to save user rating
- add a mutation to update user rating
- add a query to get user rating
- solve a lot of warning message in the console
- clean code
- add soft compare rating functionality in the rate-page radar

### Steps to test

_First scenario_ 
1. login on EOS RATE using scatter
2. go to block producer pages
3. choose one BP and click it
4. if you rating it in the past the slider component would be populated with your last rating values if is your first time all sliders should have initial value (0)
5. change the sliders values
6. click Publish Rating button (your rating should be saved or updated)

_Second scenario_ 
1. using EOS RATE without login
2. go to block producer pages
3. choose one BP and click it
4. All sliders should have initial value (0)
5. change the sliders values
6. click Publish Rating button
7. a warning message should be shown saying that you need login if you want to rate a BP
8. login using scatter
9. click again Publish Rating button

### Checklist
- [ ] I have added/updated tests to cover these changes.
- [ ] The branch name, commit history and PR title follow [conventions](https://developers.eoscostarica.io/docs/open-source-guidelines#pull-request-general-guidelines).
- [ ] I have taken into consideration any impact to site performance.

### CODE REVIEW

File: services/frontend/.env [hunk 1]
```diff
@@ -1,3 +1,4 @@
 REACT_APP_API_URL=https://api.eosio.cr
```
Comment (xavier506) [inline services/frontend/.env]: @tetogomez  The smart contract is validating ratings are only being saved for registred block producers. Sinc the contract is being tested on the jungle testnet we need to change this variable to use the jungle testnet endpoint.

https://jungle.eosio.cr

**Comment by xavier506** [comment]
@danazkari  it seems there is a pending databae migration on the server affecting this PR.
image

Also there is also a new evironmental variable that needs to be set : 

REACT_APP_EOS_API_URL=https://jungle.eosio.cr

File: services/frontend/src/models/BlockProducer/blockProducers.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+          )
+
+        this.addProducer({ ...blockProducer, data: bpData.data || [] })
+        dispatch.isLoading.storeIsContentLoading(false)
+      } catch (error) {
+        console.error('getBlockProducerByOwner', error)
+        dispatch.isLoading.storeIsContentLoading(false)
+      }
+    },
+    async getBlockProducerRatingByOwner({ bp, userAccount }, state) {
```
Comment (xavier506) [inline services/frontend/src/models/BlockProducer/blockProducers.js]: @tetogomez  Are we presenting each user with their previous ratings when they are logged in and rate a block producer?

In other words if I log in after I have already rated can I see my previous ratings so I can edit/update them?

Comment (tetogomez) [reply services/frontend/src/models/BlockProducer/blockProducers.js]: Yes !!

**Pull request merged**